src/library/autodj/autodjprocessor.cpp

```diff
@@ -1659,3 +1678,83 @@ bool AutoDJProcessor::nextTrackLoaded() {
 
     return loadedTrack == getNextTrackFromQueue();
 }
+
+double AutoDJProcessor::getFadeTime() {
+    // TODO(c3n7): Is there a better place for this? Find out.
+    // Copied from syncable.h
+    enum SyncMode {
+        SYNC_INVALID = -1,
+        SYNC_NONE = 0,
+        SYNC_FOLLOWER = 1,
+        // SYNC_MASTER_SOFT is a master that Mixxx has chosen automatically.
+        // depending on how decks stop and start, it may reassign soft master at will.
+        SYNC_MASTER_SOFT = 2,
+        // SYNC_MASTER_EXPLICIT represents an explicit request that the synacable be
+        // master. Mixxx will only remove a SYNC_MASTER_SOFT if the track is stopped or
+        // ejected.
+        SYNC_MASTER_EXPLICIT = 3,
+        SYNC_NUM_MODES
+    };
+
+    double leftBPM = m_pLeftBPM->get();
+    double rightBPM = m_pRightBPM->get();
+    int leftSyncMode = m_pLeftSyncMode->get();
+    int rightSyncMode = m_pRightSyncMode->get();
+    double fromBPM, toBPM;
+
+    double transitionTime = m_transitionTime;
+    double fadeTime;
+
+    double crossfader = getCrossfader();
+    DeckAttributes* pLeftDeck = m_decks[0];
+    DeckAttributes* pRightDeck = m_decks[1];
+
+    if (pLeftDeck->isPlaying() &&
+            (!pRightDeck->isPlaying() || crossfader < 0.0)) {
+        fromBPM = leftBPM;
+        toBPM = rightBPM;
+    } else if (pRightDeck->isPlaying()) {
+        fromBPM = rightBPM;
+        toBPM = leftBPM;
+    } else {
+        // Neither deck is playing. This state isn't reached.
+        // Use this as fallback though just in case.
+        fromBPM = leftBPM;
+        toBPM = rightBPM;
+    }
+
+    if (m_transitionUnit == TransitionUnit::BEATS) {
+        if ((leftSyncMode == SYNC_FOLLOWER && rightSyncMode == SYNC_MASTER_SOFT) ||
+                (leftSyncMode == SYNC_MASTER_SOFT && rightSyncMode == SYNC_FOLLOWER)) {
+            // Both the left and right bpm should be equal if this point is reached
+            VERIFY_OR_DEBUG_ASSERT(leftBPM == rightBPM) {
+                qDebug() << tr("leftBPM: %1   - rightBPM %2").arg(leftBPM).arg(rightBPM);
```

Comment:
this should probably be a qWarning()

Reply:
[Done. ](https://github.com/mixxxdj/mixxx/pull/2746/commits/f025f8a624e3c44a12ef8dc71e585df72eee03a5)Now using `qWarning()`.